plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -27,47 +26,38 @@
 
 import com.openshift.restclient.OpenShiftException;
 import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IPod;
 import com.openshift.restclient.model.IProject;
 import com.openshift.restclient.model.IResource;
+import com.openshift.restclient.model.IService;
 
 /**
  * Contributes OpenShift 3 specific content to the OpenShift explorer view
  * 
  * @author jeff.cantrill
  */
-public class OpenShiftExplorerContentProvider extends BaseExplorerContentProvider{
+public class OpenShiftExplorerContentProvider extends BaseExplorerContentProvider {
 	
-	private static final String [] groupings = new String [] {
-		ResourceKind.BUILD_CONFIG, 
-		ResourceKind.DEPLOYMENT_CONFIG, 
-		ResourceKind.SERVICE, 
-		ResourceKind.POD,
-		ResourceKind.REPLICATION_CONTROLLER, 
-		ResourceKind.BUILD, 
-		ResourceKind.IMAGE_STREAM, 
-		ResourceKind.ROUTE
-	};
-	
-	private Map<IProject, List<ResourceGrouping>> groupMap = new HashMap<IProject, List<ResourceGrouping>>();
+	private Map<IProject, List<Deployment>> deploymentsByProject = new HashMap<>();
+	private Map<IService, Deployment> deploymentByService = new HashMap<>();
 	
```

Comment:
We will have cases where 2 distinct object instances represent the same service, possibly in different states.  Hopefully, we have a hashing algo that will identify they are the same.

We should address the NP, but name and kind SHOULD NEVER be null since it is the fundamental THING that distinguishes the resource.